ring/ring.go

```diff
@@ -823,7 +835,11 @@ func (r *Ring) shuffleShard(identifier string, size int, lookbackPeriod time.Dur
 				if lookbackPeriod == 0 && instance.ReadOnly {
 					continue
 				}
-
+				// With lookback period >0, read only instances are only included if they have not changed read-only status in the lookback window.
```

Comment:
These two conditional statements mirror the filtering we do in `filterOutReadOnlyInstances`. It's unfortunate to do that in two places. Is there a way to filter read only instances in just 1 place?